fs/io_zenfs.cc

```diff
@@ -699,7 +699,7 @@ ZonedWritableFile::ZonedWritableFile(ZonedBlockDevice* zbd, bool _buffered,
 }
 
 ZonedWritableFile::~ZonedWritableFile() {
-  IOStatus s = zoneFile_->CloseWR();
+  IOStatus s = CloseInternal();
```

Comment:
This will trigger an unneeded metadata write in the normal case, can you make this conditional on the zone file being open for write?

Reply:
I meant that don't call closeinternal(generating another metadata sync) unless the file is still open

Reply:
Since we don't want metadata sync on the normal close path if the file is already closed, I put the check in `CloseInternal()`. I check on `zoneFile_->IsOpenForWR()` since there is no tracking in `ZonedWritableFile`.